docs/src/prototypes/dropdowns/inputWithDropdown.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+import { Dropdown, DropdownProps } from '@stardust-ui/react'
+
+import { atMentionItems, AtMentionItem } from './dataMocks'
+import { insertTextAtCursorPosition } from './utils'
+import { CustomPortal } from './CustomPortal'
+
+interface InputWithDropdownState {
+  dropdownOpen?: boolean
+  searchQuery?: string
+}
+
+const editorStyle: React.CSSProperties = {
+  backgroundColor: 'lightgrey',
+  padding: '5px',
+  minHeight: '100px',
+  outline: 0,
+}
+
+class InputWithDropdown extends React.Component<{}, InputWithDropdownState> {
+  private readonly initialState: InputWithDropdownState = {
+    dropdownOpen: false,
+    searchQuery: '',
+  }
+
+  private contendEditableRef = React.createRef<HTMLDivElement>()
+
+  state = this.initialState
+
+  render() {
+    return (
+      <>
+        <div
+          contentEditable
+          ref={this.contendEditableRef}
+          onKeyUp={this.handleEditorKeyUp}
+          style={editorStyle}
+        />
+        {this.renderDropdown()}
+      </>
+    )
+  }
+
+  private renderDropdown = () => {
+    const { dropdownOpen, searchQuery } = this.state
+
+    return (
+      <CustomPortal mountNodeId="dropdown-mount-node" open={dropdownOpen}>
+        <Dropdown
+          defaultOpen={true}
+          inline
+          search
+          items={atMentionItems}
+          toggleIndicator={null}
+          searchInput={{
+            input: { autoFocus: true, size: searchQuery.length + 1 },
+            onInputKeyDown: this.handleInputKeyDown,
+          }}
+          onSelectedChange={this.handleSelectedChange}
+          onSearchQueryChange={this.handleSearchQueryChange}
+          noResultsMessage="We couldn't find any matches."
+        />
+      </CustomPortal>
+    )
+  }
+
+  private hideDropdownAndRestoreEditor = (cb?: () => void) => {
+    this.setState(this.initialState, () => {
+      this.tryFocusEditor()
```

Comment:
```suggestion
      _.invoke(this.contendEditableRef.current, 'focus')
```

NIT: `tryFocusEditor()`  used only once, may be we can remove it?

Reply:
I think this is a matter of personal preference; I prefer keeping it more verbose, it makes it obvious to the reader what the code does without to understand details.